mlvm/mlvm.sh

```diff
@@ -0,0 +1,217 @@
+  echo "0.17.0")"
+
+readonly R_VERSION_ENV="$(R --version | sed -n 's/.*version[[:blank:]]\+\([0-9]\+\.[0-9]\).*/\1/p' )"
+
+if [[ ${R_VERSION_ENV} == "4"* ]]; then
+  R_VERSION="4.0"
+else
+  R_VERSION="3.6.0"
+fi
+
+# Pip
+BASE_PYTHON_PACKAGES=(
+  "google-cloud-bigquery==1.26.1" 
+  "google-cloud-datalabeling==0.4.0"
+  "google-cloud-storage==1.30.0"
+  "google-cloud-bigtable==1.4.0" 
+  "google-cloud-dataproc==1.0.1" 
+  "google-api-python-client==1.10.0" 
+  "matplotlib==3.3.0"
+  "mxnet==1.6.0" 
+  "nltk==3.5"
+  "numpy==1.18.4" 
+  "rpy2==3.3.3"
+  "scikit-learn==0.23.1" 
+  "scipy==1.4.1"
+  "sparksql-magic==0.0.3" 
+  "tensorflow-datasets==3.2.1"
+  "tensorflow-estimator==2.2.0"
+  "tensorflow-hub==0.8.0"
+  "tensorflow-io==0.14.0"
+  "tensorflow-probability==0.10.1" 
+  "torch==1.5.1" 
+  "torchvision==0.6.1" 
+  "xgboost==1.1.0"
+)
+
+if [[ -n ${INCLUDE_GPUS} ]]; then
+  BASE_PYTHON_PACKAGES+=("tensorflow-gpu==2.2.0")
+else
+  BASE_PYTHON_PACKAGES+=("tensorflow==2.2.0")
+fi
+
+if [ "$(echo "$DATAPROC_VERSION >= 2.0" | bc)" -eq 1 ]; then 
+  BASE_PYTHON_PACKAGES+=("spark-tensorflow-distributor==0.1.0")
+fi
+
+# Conda
+readonly BASE_R_PACKAGES=(
+  "r-essentials=${R_VERSION}"
+  "r-xgboost=0.90.0.2"
+  "r-sparklyr=1.0.0"
+)
+
+mkdir -p ${JARS_DIR}
+mkdir -p ${CONNECTORS_DIR}
+
+function execute_with_retries() {
+  local -r cmd=$1
+  for ((i = 0; i < 10; i++)); do
+    if eval "$cmd"; then
+      return 0
+    fi
+    sleep 5
+  done
+  echo "Cmd \"${cmd}\" failed."
+  return 1
+}
+
+function download_spark_jar() {
+  local -r url=$1
+
+  wget -nv --timeout=30 --tries=5 --retry-connrefused \
+  -P "${JARS_DIR}" "${url}" 
+}
+
+function download_init_actions() {
+  # Download initialization actions locally.
+  mkdir "${INIT_ACTIONS_DIR}/rapids/"
+  mkdir "${INIT_ACTIONS_DIR}/gpu/"
+  
+  gsutil -m rsync -r "${INIT_ACTIONS_REPO}/rapids/" "${INIT_ACTIONS_DIR}/rapids/"
+  gsutil -m rsync -r "${INIT_ACTIONS_REPO}/gpu/" "${INIT_ACTIONS_DIR}/gpu/"
+
+  find "${INIT_ACTIONS_DIR}" -name '*.sh' -exec chmod +x {} \;
+}
+
+function install_gpu_drivers() {
+  "${INIT_ACTIONS_DIR}/gpu/install_gpu_driver.sh"
+}
+
+function install_python_packages() {
+  local extra_python_packages
+  extra_python_packages="$(/usr/share/google/get_metadata_value attributes/PYTHON_PACKAGES || true)"
+
+  execute_with_retries "pip install ${BASE_PYTHON_PACKAGES[*]}"
+
+  if [[ -n "${extra_python_packages}" ]]; then
+    execute_with_retries "pip install ${extra_python_packages[*]}"
+  fi 
+}
+
+function install_r_packages() {  
+  local extra_r_packages 
+  extra_r_packages="$(/usr/share/google/get_metadata_value attributes/R_PACKAGES || true)"
+  
+  # Conda R channel isn't currently as up-to-date with R4 as conda-forge
+  if [[ ${R_VERSION} == "4"* ]]; then
+    channel="conda-forge"
+  else
+    channel="r"
+  fi
+
+  conda install -y -c ${channel} "${BASE_R_PACKAGES[@]}"
```

Comment:
I think it's better to use default channel - if R 4.0 will be installed, then channel should be already configured that has it.

Reply:
The R channel is a bit slower to update their packages to 4.0 than the community provides in conda-forge, in particular r-essentials does not have a 4.0 version available yet and I ran into versioning conflicts with the conda-forge 3.6 version as it's built on a newer R version than we have on Dataproc.

Reply:
I meant that we should install R 4.0 only if it already installed on the image, and if it installed on the image then it means that R 4.0 was either added to default r channel or image uses forge channel by default. In both cases we don't need to chose channel explicitly.

Reply:
I see, that makes sense. Ack.